server/controllers/users.js

```diff
@@ -0,0 +1,322 @@
+ */
+const viewUser = (req, res) => {
+  /** set default values for offset and limit */
+  let offset = 0, limit = 20;
+  if (req.query.limit || req.query.offset) {
+    req.checkQuery('limit', 'Limit must be an integer').isInt();
+    req.checkQuery('offset', 'Offset must be an integer').isInt();
+
+    returnValidationErrors(req, res);
+
+    /** convert limit and offset to number in base 10 */
+    limit = parseInt(req.query.limit, 10);
+    offset = parseInt(req.query.offset, 10);
+  }
+
+  Users.findAll({
+    include: [
+      {
+        model: Roles,
+        required: true,
+        attributes: { exclude: ['createdAt', 'updatedAt'] },
+      }
+    ],
+    offset,
+    limit,
+    attributes: { exclude: ['password', 'updatedAt', 'roleId'] },
+  })
+  .then((users) => {
+    if (users.length < 1) {
+      return res.status(200).send({
+        status: 'ok',
+        message: 'No user found',
+      });
+    }
+
+    return res.status(200).send({
+      status: 'ok',
+      users,
+    });
+  })
+  .catch(() => catchError(res));
+};
+
+/**
+ * @description gets user details using id
+ * @function getUserById
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object of matching user
+ */
+const getUserById = (req, res) => {
+  req.checkParams('id', 'No user id supplied').notEmpty();
+  req.checkParams('id', 'Only integers are allowed as user id').isInt();
+
+  returnValidationErrors(req, res);
+
+  Users.findOne({
+    where: { userId: req.params.id },
+    include: [
+      {
+        model: Roles,
+        required: true,
+        attributes: { exclude: ['createdAt', 'updatedAt'] },
+      }
+    ],
+    attributes: { exclude: ['password', 'roleId'] },
+  })
+  .then((user) => {
+    if (!user) {
+      return res.status(404).send({
+        status: 'error',
+        message: 'This user does not exist or has been previously deleted',
+      });
+    }
+
+    return res.status(200).send({
+      status: 'ok',
+      userDetails: user,
+    });
+  })
+  .catch(() => catchError(res));
+};
+
+/**
+ * @description updates user details
+ * @function updateUser
+ * @param {object} req request object
+ * @param {object} res response object
+ * @returns {object} response object containing updated user details
+ */
+const updateUser = (req, res) => {
+  req.checkParams('id', 'No user id supplied').notEmpty();
+  req.checkParams('id', 'Only integers are allowed as user id').isInt();
+
+  returnValidationErrors(req, res);
+
+  Users.findOne({
+    where: { userId: req.params.id },
+  })
+  .then((user) => {
+    if (!user) {
+      res.status(404).send({
+        status: 'error',
+        message: 'this user does not exist or has been previously deleted',
+      });
+    }
+
+    user.update({
+      fullname: req.body.fullname || user.email,
+      email: req.body.email || user.email,
+      password: passwordHash(req.body.password) || user.password,
```

Comment:
Not quite sure how the frontend is setup but usually Change password should be on it's own i.e. separate from the general profile update.